src/components/common/ChainIndicator/index.tsx

```diff
@@ -26,9 +26,11 @@ const ChainIndicator = ({ chainId, className, inline = false }: ChainIndicatorPr
     }
   }, [chainConfig])
 
+  if (!chainConfig?.chainName) return null
```

Comment:
The idea was to render the component even if the chains are still loading.
I would suggest to restore it how it was. If you want to improve it, maybe replace the `|| ' '` with a Skeleton?

Reply:
>  If you want to improve it, maybe replace the || ' ' with a Skeleton?

That wouldn't help in our scenario. Our problem with the component was that if you pass a chainId that we don't support, it will render an empty but occupying space component. Out of curiosity, why does the whitespace need to be rendered if the chains are loading?

I think if it's a hard requirement on your side, we might be better with a separate component then

Reply:
If you pass an invalid chainId, you've got bigger problems in the parent component. That's why I suggest to use `useCurrentChain` instead of your new `useChainFromQuery`. It falls back to the last chain when no valid chainId is passed.

Reply:
> If you pass an invalid chainId, you've got bigger problems in the parent component

What do you mean? On the config service, we store all chains the app supports, not only the ones we support. Of course we can filter them out but to me it seems like a waste of resources

Reply:
I explained why `useCurrentChain` doesn't suit us in the other thread. 

tl;dr the app is shared per specific chain and if no query is present, the share link is invalid

Reply:
I see now, I thought you meant invalid chain as in the Safe doesn't support it.

Reply:
I restored the previous state but added a flag not to render the whitespace if the chain isn't found